src/ShareFile/ShareFileService.php

```diff
@@ -0,0 +1,456 @@
+            ]);
+            foreach ($hearing->getChildren() as $reply) {
+                $table->addRow([
+                    ' '.$reply->name,
+                    $reply->id,
+                    $reply->progenyEditDate,
+                    json_encode($this->getMetadata($reply), JSON_PRETTY_PRINT),
+                ]);
+                foreach ($reply->getChildren() as $file) {
+                    $table->addRow([
+                        '  '.$file->name,
+                        $file->id,
+                    ]);
+                }
+            }
+        }
+
+        $table->render();
+    }
+
+    private function setMetadata(array &$item)
+    {
+        $item['_metadata'] = $this->getMetadataValues($item['Id'], ['agent_data', 'ticket_data', 'user_data']);
+    }
+
+    private function validateConfiguration()
+    {
+        $requiredFields = ['hostname', 'client_id', 'secret', 'username', 'password', 'root_id'];
+        foreach ($requiredFields as $field) {
+            if (!isset($this->configuration[$field])) {
+                throw new \RuntimeException('Configuration value "'.$field.'" missing.');
+            }
+        }
+    }
+
+    /**
+     * @param string|Item $item
+     *
+     * @return string
+     */
+    private function getItemId($item)
+    {
+        return $item instanceof Item ? $item->id : $item;
+    }
+
+    private function getChildren(string $itemId, string $type, \DateTimeInterface $changedAfter = null)
+    {
+        $query = [
+            //            '$select' => implode(',', [
+            //                'Id',
+            //                'CreationDate',
+            //                'Name',
+            // // https://community.sharefilesupport.com/citrixsharefile/topics/using-api-what-way-can-clients-listen-for-new-files?topic-reply-list[settings][filter_by]=all&topic-reply-list[settings][reply_id]=17731261#reply_17731261
+            //                'ProgenyEditDate',
+            //            ]),
+
+            //            '$orderby' => 'ProgenyEditDate asc',
+
+            //            '$expand' => implode(',', [
+            //                'Children',
+            //                'Children/Children',
+            //            ]),
+            '$filter' => 'isof(\''.$type.'\')',
+        ];
+
+        // Filter on "ProgenyEditDate" results in "500 Internal server error" in ShareFile API if non-folder items (i.e. items with no ProgenyEditDate property) exists in parent.
+//        if (null !== $changedAfter && self::SHAREFILE_FOLDER === $type) {
+//            if (isset($query['$filter'])) {
+//                $query['$filter'] .= ' and ';
+//            } else {
+//                $query['$filter'] = '';
+//            }
+//            // https://www.odata.org/documentation/odata-version-3-0/odata-version-3-0-core-protocol/#thefiltersystemqueryoption
+//            $query['$filter'] .= 'ProgenyEditDate gt date('.$changedAfter->format('Y-m-d').')';
+//        }
```

Comment:
Im assuming these are to be used later. If not just remove them.

Reply:
I'll remove it.